cmd/crates/soroban-spec-typescript/src/lib.rs

```diff
@@ -121,14 +121,28 @@ export class Client extends ContractClient {{
 }
 
 pub fn generate(spec: &[ScSpecEntry]) -> String {
-    let mut collected: Vec<_> = spec.iter().map(Entry::from).collect();
-    if !spec.iter().any(is_error_enum) {
-        collected.push(Entry::ErrorEnum {
-            doc: String::new(),
-            name: "Error".to_string(),
-            cases: vec![],
-        });
-    }
+    let collected: Vec<_> = spec.iter().map(Entry::from).collect();
+    let (error_docs, error_cases) = collected
+        .clone()
+        .into_iter()
+        .filter_map(|entry| {
+            if let Entry::ErrorEnum { doc, cases, .. } = entry {
+                Some((doc, error_cases_to_ts(&cases)))
+            } else {
+                None
+            }
+        })
+        .fold(
+            (String::new(), vec![]),
+            |(acc_docs, acc_cases), (docs, cases)| {
+                (format!("{acc_docs}{docs}"), [acc_cases, cases].concat())
```

Comment:
Generally this is a bad way to concatenate strings. Basically you it's reallocating and copying each time. The better way is with a join after collecting all the strings. I doubt it will make much of an issue here given how few errors there are, but a good thing to know in general.

Reply:
Is this what you're picturing?

`rs
.fold(
    (vec![], vec![]),
    |(acc_docs, acc_cases), (docs, cases)| {
        (
            [acc_docs, [docs]].concat(),
            [acc_cases, cases].concat(),
        )
    }
);
let error_docs = doc_to_ts_doc(&error_docs.join(""), None, 0);
```

Reply:
Updated the code; lmk what you think

Reply:
Maybe, it's hard for me mentally understand what the first element in the tuple is. If you need you might need to `.flatten` first.